futures-util/src/stream/select_with_strategy.rs

```diff
@@ -14,6 +13,13 @@ pub enum PollNext {
     Right,
 }
 
+enum PollSide {
```

Comment:
This is identical to `PollNext`, but we're using it to represent the last polled side (not the next one).

Would you be open to just renaming `PollNext` to `PollSide`? I think the `Next` suffix only reduces the contexts it can be used in. It would be a breaking change, but that might be acceptable for 0.4.0?